design/census-import/generate-census-records.js

```diff
@@ -26,53 +26,78 @@ function writeCsv (data) {
   const headers = ['LEA', 'Estab', 'UPN', 'Surname', 'Forename', 'Middlenames', 'Gender', 'DOB']
   const ws = fs.createWriteStream(outputFilename, { flags: 'a' })
   csv
-    .write(data, {headers: headers})
+    .write(data, { headers: headers })
     .pipe(ws)
 }
 
-async function getSchoolsLength () {
-  const sql = `SELECT COUNT(*) AS [cnt]
+async function getSchoolDfeNumbers () {
+  const sql = `SELECT dfeNumber
   FROM ${sqlService.adminSchema}.school`
   const result = await sqlService.query(sql, [])
-  const obj = R.head(result)
-  return R.prop('cnt', obj)
+  return R.map(r => R.prop('dfeNumber', r), result)
+}
+
+async function getSchoolDfeNumberCount () {
+  const sql = `SELECT COUNT(dfeNumber) AS cnt
+  FROM ${sqlService.adminSchema}.school`
+  const result = await sqlService.query(sql, [])
+  return R.prop('cnt', R.head(result))
+}
+
+function generateUpn (dfeNumber) {
+  const leaCode = dfeNumber.slice(0, 3)
+  const estabCode = dfeNumber.slice(3, dfeNumber.length)
+  const minYearValue = academicYear - 11
+  const maxYearValue = academicYear - 7
+  const year = Math.floor(Math.random() * (maxYearValue - minYearValue + 1) + minYearValue).toString().substr(-2)
+  let baseUpn = leaCode + estabCode + year
+  pupilCount += 1
+  if (pupilCount > 999) {
+    pupilCount = 1
+    baseUpn = (parseInt(baseUpn) + 1000).toString()
+  }
+  const serial = pupilCount.toString().padStart(3, '0')
+  return upnService.calculateCheckLetter(baseUpn + serial) + baseUpn + serial
 }
 
-async function main (options) {
-  winston.info('Generating census records...')
-  const { recordsLength } = options
+async function main () {
+  await sqlService.initPool()
+  console.log('Generating census records...')
   const csvData = []
-  let baseUpn = '702500001'
-  let schoolsLength
+  let dfeNumbers
+  let dfeNumberCount
+  let upns = []
   try {
-    schoolsLength = await getSchoolsLength()
-    schoolsLength = schoolsLength - 1
+    dfeNumbers = await getSchoolDfeNumbers()
+    dfeNumberCount = await getSchoolDfeNumberCount()
```

Comment:
seems pointless, why not just `dfeNumbers.length` ?

Reply:
@GuyHarwood indeed, thanks that was totally redundant. updated